### PR TITLE
feat: Add azure resource manager to replace arm template 

### LINKED ISSUE #3385: Add ability to provision Azure hosting environments
Add the ability to create the cloud resources necessary for operating a bot to Composer. After logging into Azure, the user should be able to create all of the necessary resources and automatically create a publishing profile to use those resources. 

## Describe the solution you'd like

- [ ] Need design input from @sangwoohaan and @mareekuh 
- [ ] In the publish page, add a button for “Create Hosting Environment” that, when clicked, leads to a UI used to choose/configure a set of resources to create
- [ ] Upon submission, carry out the provisioning process. This will require a clear progress indicator in the UI, and will need a lot of error handling and error correction as the provisioning process can take a long time and also is very error prone. Luckily many of the error scenarios are already known and handled in the existing CLI script. 
- [ ] Upon success, a new publishing profile is added to the bot representing the newly created environment 

We should build this with a few things in mind:
* In the future we want ability to progressively add resources to a configuration
* In the future we want ability to choose from existing resources to construct a configuration

## Additional context

Requires Azure login: #3384

### PR DESCRIPTION
**Pull request by zidaneymar**
add azure resource manager to handle the breaking down for arm template

## Description
1. Arm template break down
2. Add AzureResourceManager

## Task Item
fixes #3385
part of #3385, add more concrete controls on the resources

### CODE REVIEW

**Comment by zidaneymar** [comment]
Need to enable azure web login.

File: Composer/plugins/lib/bot-deploy/src/azureResourceManager/azureResourceManagerConfig.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  // Create bot service
+  bot: BotConfig;
+
+  // Create web app
+  webApp: WebAppConfig;
+
+  // Create qna resource
+  qnaResource: QnAResourceConfig;
+
+  // Deployments counter
```
[the remark this answers is not in the document]
Comment (zidaneymar) [reply Composer/plugins/lib/bot-deploy/src/azureResourceManager/azureResourceManagerConfig.ts:43]: > What does "counter" mean in this context?

Counter is a resource, could be used as a counter for "how many deployments has been made by users", same as the resource in arm template file:https://github.com/microsoft/BotFramework-Composer/blob/9cbec03bd75c697814a5dff7adcc1e6ea63adae7/Composer/plugins/samples/assets/shared/scripts/DeploymentTemplates/template-with-preexisting-rg.json#L189

File: Composer/plugins/lib/bot-deploy/src/azureResourceManager/azureDeploymentOutput.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
```
[the remark this answers is not in the document]
Comment (zidaneymar) [reply Composer/plugins/lib/bot-deploy/src/azureResourceManager/azureDeploymentOutput.ts:1]: > General feedback on the interfaces here: why is everything optional? Is this really the case?

Sorry for replying late, optional just make it easier for initializing, I could make everything required.

[the remark this answers is not in the document]
Comment (cwhitten) [reply Composer/plugins/lib/bot-deploy/src/azureResourceManager/azureDeploymentOutput.ts]: @zidaneymar if everything is required for working functionality, then yes they should be required--can you update all interfaces to reflect this please?

**Comment by cwhitten** [comment]
This change needs tests

**Pull request closed without merging**